app/src/main/java/com/andrew/tasky/presentation/agenda_item_detail/AgendaItemDetailFragment.kt

```diff
@@ -334,18 +350,44 @@ class AgendaItemDetailFragment: Fragment(R.layout.fragment_agenda_item_detail) {
         popupMenu.show()
     }
 
+    private fun getAgendaItemCurrentInfo(): AgendaItem {
+        return AgendaItem(
+            agendaItemType,
+            viewModel.isDone.value,
+            viewModel.title.value,
+            viewModel.description.value,
+            LocalDateTime.of(viewModel.selectedDate.value, viewModel.selectedTime.value),
+            reminderTime = viewModel.selectedReminderTime.value
+        )
+    }
+
+    private fun saveAgendaItem(){
+        val newAgendaItem = getAgendaItemCurrentInfo()
+        if(args.agendaItem != null) {
+            args.agendaItem?.let { oldAgendaItem ->
+                AgendaItems.replaceAgendaItem(newAgendaItem,oldAgendaItem)
+            }
+        }
+        else{
+            AgendaItems.addAgendaItem(newAgendaItem)
+        }
+    }
```

Comment:
Same for this function. The fragment should only contain logic to observe UI state and change the UI based on that.